.github/workflows/ci.yaml

```
@@ -34,3 +34,7 @@ jobs:
         uses: guardian/actions-merge-release-changes-to-protected-branch@v1.1.1
         with:
           github-token: ${{ secrets.GITHUB_TOKEN }}
+          # `guardian/actions-merge-release-changes-to-protected-branch` doesn't support v2 of `package-lock.json`
+          # We need to update it.
+          # In the meantime, allow all changes.
+          additional-changes: '{"package-lock.json": ["*"], ".github/workflows/ci.yaml": ["*"]}'
```

[Here](https://github.com/guardian/cdk/pull/864/files) is an update to the v1 lockfile. There are more changes to the lockfile in this PR because it's in v2 format.

This configures `guardian/actions-merge-release-changes-to-protected-branch` to allow any change to `package-lock.json` as a short term fix, until `guardian/actions-merge-release-changes-to-protected-branch` supports the v2 lockfile.

`.github/workflows/ci.yaml` is added to allow us to change this file in this PR. We should remove it in a follow up PR.

---

I think it maybe has to be `"*"` rather than `["*"]`..?

```suggestion
          additional-changes: '{"package-lock.json": "*", ".github/workflows/ci.yaml": "*"}'
```

---

That doesn't work either 😢

---

Ah, I think this must be why:

https://github.com/guardian/actions-merge-release-changes-to-protected-branch/blob/70bdbc3cfee6c5c097620c9c55a9057d257591e0/src/config.ts#L87

It turns out `additional-changes` can only be used to specify changes to other files, as well as the expected files dependent on which package manager is used.

---

I think it would be relatively easy to add a new `npm >= 8` option for the package manager config item that expected 4 changes to the package lock if you wanted to pair on doing that this afternoon?

https://github.com/guardian/actions-merge-release-changes-to-protected-branch/blob/70bdbc3cfee6c5c097620c9c55a9057d257591e0/src/config.ts#L20-#L28

---

Had a few meetings. Let's try tomorrow?

---

https://github.com/guardian/actions-merge-release-changes-to-protected-branch/pull/218